enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreState.java

```diff
@@ -238,14 +234,14 @@ public synchronized void start() throws IOException, RaftLogCompactedException,
     @Override
     public synchronized void stop() throws Throwable
     {
-        if ( applier.sync( true ) )
-        {
-            flush();
-        }
+        applier.sync( true );
+        flush();
     }
 
-    public synchronized CoreSnapshot snapshot() throws IOException, RaftLogCompactedException
+    public synchronized CoreSnapshot snapshot() throws IOException, RaftLogCompactedException, InterruptedException
     {
+        applier.sync( false );
+
```

Comment:
no baker, it's just not visible on the interface... this might mean several things, including that it might be refactored into smaller testable components... I am not intending on doing that right now